services/net/filemonitor/FilemonitorAction.cs

```diff
@@ -88,6 +98,129 @@ public FileMonitorAction(IHttpRequestClient httpClient, IApiService api, IKafkaM
+            }
+
+            client.Disconnect();
+        }
+        else
+        {
+            _logger.LogError("SSH Private key file does not exist: {file}", sshKeyFile);
+        }
+    }
+
+    /// <summary>
+    /// Fetch a list of file names from the remote data source based on configuration.
+    /// </summary>
+    /// <param name="client"></param>
+    /// <param name="remoteFullName"></param>
+    /// <returns></returns>
+    private static async Task<IEnumerable<SftpFile>> FetchFileListing(SftpClient client, string remoteFullName)
+    {
+        // TODO: Fetch file from source data location.  Only continue if the image exists.
+        return await Task.Factory.FromAsync<IEnumerable<SftpFile>>((callback, obj) => client.BeginListDirectory(remoteFullName, callback, obj), client.EndListDirectory, null);
+    }
+
+    /// <summary>
+    /// Copy a file for processing from the remote server
+    /// </summary>
+    /// <param name="client"></param>
+    /// <param name="ingest"></param>
+    /// <param name="pathToFile"></param>
+    /// <returns></returns>
+    private async Task CopyFile(SftpClient client, IngestModel ingest, string pathToFile)
+    {
+        var outputPath = GetMonitorPath(ingest);
+        var fileName = Path.GetFileName(pathToFile);
+        var outputFile = Path.Combine(outputPath, fileName);
+
+        if (!System.IO.File.Exists(outputFile))
+        {
+            if (!Directory.Exists(outputPath))
+            {
+                Directory.CreateDirectory(outputPath);
+            }
+            using var saveFile = File.OpenWrite(outputFile);
+            var task = Task.Factory.FromAsync(client.BeginDownloadFile(pathToFile, saveFile), client.EndDownloadFile);
+            await task;
+        }
+    }
+
+    /// <summary>
+    /// Get the output path to store the file.
+    /// </summary>
+    /// <param name="ingest"></param>
+    /// <returns></returns>
+    protected string GetMonitorPath(IngestModel ingest)
+    {
+        return Path.Combine(this.Options.VolumePath, ingest.DestinationConnection?.GetConfigurationValue("path")?.MakeRelativePath() ?? "", $"{ingest.Source?.Code}/{GetLocalDateTime(ingest, DateTime.Now):yyyy-MM-dd/}");
+    }
+
+    /// <summary>
+    /// Get the path to the source file.
+    /// </summary>
+    /// <param name="ingest"></param>
+    /// <returns></returns>
+    protected string GetInputPath(IngestModel ingest)
+    {
+        return Path.Combine(ingest.SourceConnection?.GetConfigurationValue("path") ?? "", $"{GetLocalDateTime(ingest, DateTime.Now):yyyy/MM/dd/}");
+    }
+
+    /// <summary>
+    /// Convert to timezone and return as local.
+    /// Dates should be stored in the timezone of the data source.
+    /// </summary>
+    /// <param name="ingest"></param>
+    /// <param name="date"></param>
+    /// <returns></returns>
+    protected DateTime GetLocalDateTime(IngestModel ingest, DateTime date)
```

Comment:
We need to at some point make this a shared function as we use it in a few places now.  Probably belongs in a parent class.  You don't need to change it now though.